### PR TITLE
fix(ui): add null-safe handling for assignedWindows in status panel

### LINKED ISSUE #22786: argocd Ui is having an issue while re loading
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

Something went wrong!

Consider submitting an issue [here](https://github.com/argoproj/argo-cd/issues/new?labels=bug&template=bug_report.md).

Stacktrace:

TypeError: Cannot read properties of null (reading 'assignedWindows')
    at Object.children (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:1571950)
    at Rr.render (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:1290403)
    at Bi (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:581352)
    at zi (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:581148)
    at Es (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:616684)
    at kl (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:608167)
    at xl (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:608092)
    at hl (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:605125)
    at https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:556845
    at t.unstable_runWithPriority (https://corecd-prd.centralus.chrazure.cloud/main.324c84a52f91c71095c3.js:2:631950)

**To Reproduce**

Login into argocd and wait for few minutes with out any operation refresh the page it will loose the connectivity and will the following  error  along with info that i have pasted above 

Unable to load data: Request has been terminated Possible causes: the network is offline, Origin is not allowed by Access-Control-Allow-Origin, the page is being unloaded, etc.

**Expected behavior**

**Screenshots**

Image

**Version** 

```shell
v2.12.8+9c3b45f
```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by choejwoo**
Closes #22786

Hello, I checked this issue and found that the problem occurs when the `assignedWindows` reference is unavailable.
After reviewing the code, I identified the parts where `assignedWindows` is being accessed and manually injected a null value to confirm that an error occurs when it’s referenced.
#### Without null-safe handling
null_guard_assignedWindows

Since `assignedWindows` is used only in a few places, I added null-safe handling where applicable.
After adding the null-safe handling, I tested again by forcing a null value, and confirmed that the UI loaded properly without any errors.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Title of the PR](https://argo-cd.readthedocs.io/en/latest/developer-guide/submit-your-pr/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by rumstead — APPROVED** [review]

File: ui/src/app/applications/components/utils.tsx [hunk 1]
```diff
@@ -1570,13 +1570,15 @@ export const SyncWindowStatusIcon = ({state, window}: {state: appModels.SyncWind
...
+export const ApplicationSyncWindowStatusIcon = ({project, state}: {project: string; state?: appModels.ApplicationSyncWindowState}) => {
     let className = '';
     let color = '';
     let deny = false;
     let allow = false;
     let inactiveAllow = false;
-    if (state.assignedWindows !== undefined && state.assignedWindows.length > 0) {
+    if (!state) {
+        allow = true;
+    } else if (state.assignedWindows !== undefined && state.assignedWindows.length > 0) {
```
Comment (blakepettersson) [inline ui/src/app/applications/components/utils.tsx]: Why not 

```suggestion
    if (state?.assignedWindows !== undefined && state?.assignedWindows.length > 0) {
```

**Review by blakepettersson — APPROVED** [review]
LGTM, thanks!

**Comment by blakepettersson** [comment]
@choejwoo does this need to be backported? To which versions if so? I'll add a cherry-pick for 3.2

**Comment by choejwoo** [comment]
@blakepettersson Thank you for the quick response and review!
I’m not completely sure, but according to the issue, it is occured in the 2.12 version.
If needed, I think it would be good to apply the fix to the recent LTS versions as well.
Thanks!
